resource_calendar_flexible_hours_exclude_weekend/models/resource_calendar.py

```diff
@@ -0,0 +1,142 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+from collections import defaultdict
+from datetime import timedelta
+
+from dateutil.rrule import DAILY, rrule
+from pytz import timezone
+
+from odoo import fields, models
+
+from odoo.addons.hr_work_entry_contract.models.hr_work_intervals import WorkIntervals
+
+_logger = logging.getLogger(__name__)
+
+
+class ResourceCalendar(models.Model):
+    _inherit = "resource.calendar"
+
+    exclude_weekends = fields.Boolean()
+
+    # Override to return weekends as special days if exclude_weekends is set
+    def _attendance_intervals_batch(
+        self, start_dt, end_dt, resources=None, domain=None, tz=None, lunch=False
+    ):
+        """
+        Override to adjust start_dt if it falls on a weekend.
+        - Convert start_dt to the relevant timezone
+        - If Saturday or Sunday → move to next Monday
+        - Then call super() with the adjusted start_dt
+        """
+        # Ensure timezone awareness
+        assert start_dt.tzinfo and end_dt.tzinfo, "Datetimes must be timezone-aware"
+        self.ensure_one()
+
+        if not resources:
+            resources = self.env["resource.resource"]
+            resources_list = [resources]
+        else:
+            resources_list = list(resources) + [self.env["resource.resource"]]
+
+        resources_with_flex_no_weekend = []
+        other_resources = []
+        for resource in resources_list:
+            if (
+                resource
+                and resource.calendar_id.flexible_hours
+                and resource.calendar_id.exclude_weekends
+            ) or (self.flexible_hours and self.exclude_weekends):
+                resources_with_flex_no_weekend.append(resource)
+            else:
+                other_resources.append(resource)
+        if other_resources:
+            res_others = super()._attendance_intervals_batch(
+                start_dt,
+                end_dt,
+                sum(other_resources, self.env["resource.resource"]),
+                domain,
+                tz,
+                lunch,
+            )
+        else:
+            # containing resources (empty or not)
+            res_others = {}
+        if resources_with_flex_no_weekend:
+            skipping_res = defaultdict(dict)
+            resources_per_tz = defaultdict(list)
+            for resource in resources_with_flex_no_weekend:
+                resources_per_tz[tz or timezone((resource or self).tz)].append(resource)
+            for tz, tz_resources in resources_per_tz.items():
+                skipping_start_dt = start_dt.astimezone(tz)
+                # 0=Monday, 6=Sunday
+                weekday = skipping_start_dt.weekday()
+                if weekday in (5, 6):  # Saturday or Sunday
+                    # Move to next Monday, midnight in the considered TZ
+                    days_to_monday = 7 - weekday
+                    skipping_start_dt += timedelta(days=days_to_monday)
```

Comment:
```suggestion
                    skipping_start_dt += timedelta(days=days_to_monday, hours=1)
```

Handle the fact that there could be a change from non DST to DST period in the interval